src/main/java/org/openmrs/maven/plugins/SetupPlatform.java

```diff
@@ -0,0 +1,181 @@
+     */
+    private String serverId;
+
+    /**
+     * Platform version
+     *
+     * @parameter expression="${version} default-value="1.11.2"
+     */
+    private String version;
+
+    /**
+     * DB Driver type
+     *
+     * @parameter expression="${dbDriver}
+     */
+    private String dbDriver;
+
+    /**
+     * DB Uri
+     *
+     * @parameter expression="${dbUri}
+     */
+    private String dbUri;
+
+    /**
+     * DB User
+     *
+     * @parameter expression="${dbUser}
+     */
+    private String dbUser;
+
+    /**
+     * DB Pass
+     *
+     * @parameter expression="${dbPassword}
+     */
+    private String dbPassword;
+
+    /**
+     * Component for user prompt
+     *
+     * @component
+     */
+    private Prompter prompter;
+
+    /**
+     * The Maven BuildPluginManager component.
+     *
+     * @component
+     * @required
+     */
+    private BuildPluginManager pluginManager;
+
+    public void execute() throws MojoExecutionException {
+        // check if user not set serverId parameter
+        if (null == serverId) try {
+            // prompt this param
+            serverId = prompter.prompt("Please specify server id");
+        } catch (PrompterException e) {
+            e.printStackTrace();
+        }
+        // path to omrs home
+        String omrsHome = System.getProperty("user.home") + File.separator + SDKValues.OPENMRS_SERVER_PATH;
+        // path to server with serverId
+        File serverPath = new File(omrsHome + File.separator + serverId);
+        // check existence
+        if (serverPath.exists()) {
+            // show massage and exit
+            getLog().error("Server with same id already created");
+            return;
+        }
+        executeMojo(
+                plugin(
+                        groupId(SDKValues.ARCH_GROUP_ID),
+                        artifactId(SDKValues.ARCH_ARTIFACT_ID),
+                        version(SDKValues.ARCH_VERSION)
+                ),
+                goal("generate"),
+                configuration(
+                        element(name("archetypeCatalog"), SDKValues.ARCH_CATALOG),
+                        element(name("interactiveMode"), interactiveMode),
+                        //element(name("package"), SDKValues.PROJECT_PACKAGE),
+                        element(name("archetypeGroupId"), SDKValues.ARCH_PROJECT_GROUP_ID),
+                        element(name("archetypeArtifactId"), SDKValues.ARCH_PROJECT_ARTIFACT_ID),
+                        element(name("archetypeVersion"), SDKValues.ARCH_PROJECT_VERSION)
+                        //element(name("groupId"), SDKValues.PROJECT_GROUP_ID),
+                        //element(name("artifactId"), serverId)
+                ),
+                executionEnvironment(mavenProject, mavenSession, pluginManager)
+        );
+        // move server after creating
+        File currentProject = new File(System.getProperty("user.dir") + File.separator + serverId);
```

Comment:
Did you know that you can do new File(System.getProperty("user.dir"), serverId)? It's cleaner and you don't have to worry about doubled separators.